apps/web-giddh/src/app/vouchers/preview/preview.component.ts

```diff
@@ -1,63 +1,1295 @@
+                    hasPrinted = true;
+                    this.thermalService.print(this.defaultThermalTemplate, response);
+                }
+            } else {
+                this.store.dispatch(this.invoiceReceiptActions.getVoucherDetailsV4(this.selectedInvoice?.account?.uniqueName ?? this.selectedInvoice?.vendor?.uniqueName, {
+                    invoiceNumber: this.selectedInvoice?.voucherNumber,
+                    voucherType: this.voucherType,
+                    uniqueName: this.selectedInvoice?.uniqueName
+                }));
+            }
+        });
+    }
+
+    /**
+     * Handle Edit voucher redirect to voucher edit page with respective voucher
+     *
+     * @memberof VouchersPreviewComponent
+     */
+    public editVoucher(): void {
+        const queryParams = {
+            from: this.advanceFilters.from,
+            to: this.advanceFilters.to,
+            page: this.advanceFilters.page
+        }
+
+        if (this.voucherType === VoucherTypeEnum.generateEstimate) {
+            this.router.navigate([`/pages/vouchers/estimates/${this.selectedInvoice?.account?.uniqueName}/${this.selectedInvoice?.voucherNumber}/edit`], { queryParams: queryParams });
+        } else if (this.voucherType === VoucherTypeEnum.generateProforma) {
+            this.router.navigate([`/pages/vouchers/proformas/${this.selectedInvoice?.account?.uniqueName}/${this.selectedInvoice?.voucherNumber}/edit`], { queryParams: queryParams });
+        } else {
+            this.router.navigate([`/pages/vouchers/${this.urlVoucherType}/${this.selectedInvoice?.account?.uniqueName ?? this.selectedInvoice?.vendor?.uniqueName}/${this.selectedInvoice?.uniqueName}/edit`], { queryParams: queryParams });
+        }
+    }
+
+    /**
+     * Handle Voucher Actions API Call
+     *
+     * @param {*} voucher
+     * @param {string} action
+     * @memberof VouchersPreviewComponent
+     */
+    public actionVoucher(action: string, event?: any): void {
+        if (action) {
+            if (action === 'open') {
+                this.componentStore.purchaseOrderStatusUpdate({ accountUniqueName: this.selectedInvoice?.vendor?.uniqueName, payload: { action: action, purchaseNumber: this.selectedInvoice?.voucherNumber } });
+            } else {
+                this.componentStore.actionVoucher({ voucherUniqueName: this.selectedInvoice?.uniqueName, payload: { action: action, voucherType: this.voucherType } });
+            }
+        } else {
+            this.componentStore.actionVoucher({ voucherUniqueName: event?.uniqueName, payload: event });
+        }
+    }
+
+    /**
+     * Copy Invoice and redirect to respective page
+     *
+     * @memberof VouchersPreviewComponent
+     */
+    public copyInvoice(): void {
+        if (this.voucherType === VoucherTypeEnum.purchase) {
+            this.router.navigate([`/pages/proforma-invoice/invoice/purchase/${this.selectedInvoice?.account?.uniqueName}/${this.selectedInvoice?.uniqueName}/copy`]);
```

Comment:
voucher version 1 url???

Reply:
We are using old  code and url for copy invoice for now.